drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#define DT_DRV_COMPAT nxp_irtc
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/irq.h>
+
+struct nxp_irtc_config {
+	RTC_Type *base;
+	uint8_t alarm_match_flag;
+	uint8_t clock_select;
+	bool is_output_clock_enabled;
+	void (*irq_config_func)(const struct device *dev);
```

Comment:
reorder the struct members to be like:
base
irq_config_func
is_output_clock_enabled
clock_select
alarm_match_flag

this should make the struct smaller by putting the uint8_t at the bottom